pkg/descheduler/strategies/pod_antiaffinity.go

```diff
@@ -49,22 +49,25 @@ func removePodsWithAffinityRules(client clientset.Interface, policyGroupVersion
 		}
 		totalPods := len(pods)
 		for i := 0; i < totalPods; i++ {
-			if checkPodsWithAntiAffinityExist(pods[i], pods) {
-				success, err := evictions.EvictPod(client, pods[i], policyGroupVersion, dryRun)
-				if !success {
-					glog.Infof("Error when evicting pod: %#v (%#v)\n", pods[i].Name, err)
-				} else {
-					podsEvicted++
-					glog.V(1).Infof("Evicted pod: %#v (%#v)\n because of existing anti-affinity", pods[i].Name, err)
-					// Since the current pod is evicted all other pods which have anti-affinity with this
-					// pod need not be evicted.
-					// Update pods.
-					pods = append(pods[:i], pods[i+1:]...)
-					i--
-					totalPods--
+			if nodePodCount[node]+1 <= maxPodsToEvict {
```

Comment:
Nit: I prefer more:

```go
if nodePodCount[node]+1 > maxPodsToEvict {
     break
}
```

rather than to move entire block. It looks better in the diff and gives assumptions valid till the end of the for block.